images/router/haproxy/conf/haproxy-config.template

```diff
@@ -48,56 +48,27 @@ defaults
   #server openshift_backend 127.0.0.1:8080
   errorfile 503 /var/lib/haproxy/conf/error-page-503.http
 
-{{ if (matchPattern "[1-9][0-9]*(us|ms|s|m|h|d)?" (env "ROUTER_DEFAULT_CONNECT_TIMEOUT" "")) }}
-  timeout connect {{env "ROUTER_DEFAULT_CONNECT_TIMEOUT" "5s"}}
-{{ else }}
-  timeout connect 5s
-{{ end }}
-{{ if (matchPattern "[1-9][0-9]*(us|ms|s|m|h|d)?" (env "ROUTER_DEFAULT_CLIENT_TIMEOUT" "")) }}
-  timeout client {{env "ROUTER_DEFAULT_CLIENT_TIMEOUT" "30s"}}
-{{ else }}
-  timeout client 30s
-{{ end }}
-{{ if (matchPattern "[1-9][0-9]*(us|ms|s|m|h|d)?" (env "ROUTER_DEFAULT_SERVER_TIMEOUT" "")) }}
-  timeout server {{env "ROUTER_DEFAULT_SERVER_TIMEOUT" "30s"}}
-{{ else }}
-  timeout server 30s
-{{ end }}
-{{ if (matchPattern "[1-9][0-9]*(us|ms|s|m|h|d)?" (env "ROUTER_SLOWLORIS_TIMEOUT" "")) }}
-  timeout http-request {{env "ROUTER_SLOWLORIS_TIMEOUT" "10s" }}
-{{ else }}
-  timeout http-request 10s
-{{ end }}
-{{ if (matchPattern "[1-9][0-9]*(us|ms|s|m|h|d)?" (env "ROUTER_SLOWLORIS_HTTP_KEEPALIVE" "")) }}
-  timeout http-keep-alive {{env "ROUTER_SLOWLORIS_HTTP_KEEPALIVE" "" }}
-{{ else }}
-  timeout http-keep-alive 300s
-{{ end }}
+  timeout connect {{firstMatch "[1-9][0-9]*(us|ms|s|m|h|d)?" (env "ROUTER_DEFAULT_CONNECT_TIMEOUT" "") "5s"}}
+  timeout client {{firstMatch "[1-9][0-9]*(us|ms|s|m|h|d)?" (env "ROUTER_DEFAULT_CLIENT_TIMEOUT" "") "30s"}}
+  timeout server {{firstMatch "[1-9][0-9]*(us|ms|s|m|h|d)?" (env "ROUTER_DEFAULT_SERVER_TIMEOUT" "") "30s"}}
+  timeout http-request {{firstMatch "[1-9][0-9]*(us|ms|s|m|h|d)?" (env "ROUTER_SLOWLORIS_TIMEOUT" "") "10s"}}
+  timeout http-keep-alive {{firstMatch "[1-9][0-9]*(us|ms|s|m|h|d)?" (env "ROUTER_SLOWLORIS_HTTP_KEEPALIVE" "") "300s"}}
 
   # Long timeout for WebSocket connections.
-{{ if (matchPattern "[1-9][0-9]*(us|ms|s|m|h|d)?" (env "ROUTER_DEFAULT_TUNNEL_TIMEOUT" "")) }}
-  timeout tunnel {{env "ROUTER_DEFAULT_TUNNEL_TIMEOUT" "1h" }}
-{{ else }}
-  timeout tunnel 1h
-{{ end }}
+  timeout tunnel {{firstMatch "[1-9][0-9]*(us|ms|s|m|h|d)?" (env "ROUTER_DEFAULT_TUNNEL_TIMEOUT" "") "1h"}}
 
-{{ if (matchPattern "true|TRUE" (env "ROUTER_ENABLE_COMPRESSION" "false")) }}
+{{ if isTrue (env "ROUTER_ENABLE_COMPRESSION" "false") }}
```

Comment:
Would we need to change the documentation because of the way isTrue is implemented using strconv? The amount of acceptable values increases  1, t, T, TRUE, true, True